src/BitcoinD.cpp

```diff
@@ -427,6 +427,27 @@ void BitcoinDMgr::refreshBitcoinDZmqNotifications()
                         badFormat = true;
                         continue; // skip this one -- it will likely confuse libzmq
                     }
+                } else if (addr.startsWith("ipc://") || addr.startsWith("unix:")) {
+                    DebugM("getzmqnotifications: Unix domain socket address: ", addr);
+
+                    const QString socketPath = addr.startsWith("ipc://")
+                        ? addr.mid(6)
+                        : addr.mid(5);
+
+                    try {
+                        if (socketPath.isEmpty()) {
+                            throw std::runtime_error("Socket path is empty.");
+                        }
+                        // Check if the socket exists
+                        QFileInfo socketInfo(socketPath);
+                        if (!socketInfo.exists()) {
+                            throw std::runtime_error("Socket does not exist: " + socketPath.toStdString());
+                        }
+                    } catch (const std::exception &e) {
```

Comment:
Oh hey man I like that you respected the coding style and way of doing things of this codebase. Code looks good and these changes should work. I will test them today.

Only caveat here is I'm not sure how comfortable I am with "flow control via exceptions".. but maybe I'll let it slide in this one case.  The if() block above that inspired you here actually is calling `Util::PartsHostPortPair` that reports errors via exceptions.. hence why it had a try / catch. Whereas here .. technically one could just set some "errMsg" variable to accomplish the same thing... and then check at the end `if (!errMsg.isEmpty())` or some such pattern.

Anyway I'll test this and merge if it works.  Great!